java-client-generated/src/main/java/org/ergoplatform/restapi/client/TransactionsApi.java

```diff
@@ -77,6 +77,37 @@ Call<Transactions> getUnconfirmedTransactions(
         @retrofit2.http.Query("limit") Integer limit                ,     @retrofit2.http.Query("offset") Integer offset                
   );
 
+  /**
+   * Get unconfirmed transaction from pool. Available from node version 4.0.105 and up. Older
+   * versions will return a 404 error, as will newer versions when the transaction is not in
+   * mempool.
```

Comment:
Using 404 error doesn't looks like a good idea to me. 
It is better to return successfully even if there was no transaction in the pool.
Otherwise hard to distinguish between network problem and just absence of tx.
The first should be troubleshooted, the later maybe wait or something.

Reply:
We just agreed with @kushti that I change the Node's `RejectionHandler` to return `400` instead of `404` for querying non-existing endpoint ... as currently we use `404` when we ask for a resource like `TX` and it is not present... We cannot return `200` as there is nothing to put into Body. And there is a lot of these `Option[T]` endpoints

Reply:
That is good news. However, the problem will persist that old nodes still return 404 as do nodes when the transaction does not exist. This problem will become less important in the future when newer node versions are spreading.

Reply:
> agreed with @kushti that I change the Node's `RejectionHandler` to return `400`

@pragmaxim so this will fix all the enpoints, not just this specific one?

Reply:
@aslesarenko yes it is done at RejectionHandler level https://github.com/ergoplatform/ergo/pull/1882/files#diff-7f87a6eb6aa8d2abbbb020080fa251cfb6b36afa976ab2c869766ffe92fe42fdR34

Reply:
@MrStahlfelge so, with that difference on 400 and 404, are any changes need to be done here. Maybe update doc comments?

Reply:
No changes to the doc needed: New node versions support the endpoint and will return 404 if transaction is not available or 2xx if it is. Old endpoints will return 404 in any case. We only solved 404 clashing the problem for future endpoints to be added.